### PR TITLE
Handle quota exceeded and running low on disk space errors while saving a key in localStorage

### LINKED ISSUE #2884: NS_ERROR_FILE_NO_DEVICE_SPACE: No error message
Sentry Issue: [APPSMITH-HQ](https://sentry.io/organizations/appsmith/issues/2200490431/?referrer=github_integration)

```
NS_ERROR_FILE_NO_DEVICE_SPACE: No error message
  at getEnvLogLevel (utils/AppsmithUtils.tsx:165:42)
  at appInitializer (utils/AppsmithUtils.tsx:86:16)
  at 1095 (index.tsx:26:1)
  at f (/applications:1:1165)
  at 812 (None:233:16)
...
(4 additional frame(s) were not displayed)
```

### PR DESCRIPTION
**Pull request by rishabhsaxena**
## Description
show better error messages if running out of quota or disk space for localStorage

Fixes #2884

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/utils/localStorage.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+import { Variant } from "components/ads/common";
+import { Toaster } from "components/ads/Toast";
+
+const LOCAL_STORAGE_QUOTA_EXCEEDED_MESSAGE =
```
Comment (mohanarpit) [inline app/client/src/utils/localStorage.tsx:4]: Please move this to `messages.ts` file. Having all user-facing messages in a single file helps us improve the language used easily. Also, in the future, it makes internationalization also much easier. 

Hunting for messages across the code base is really hard.

File: app/client/src/constants/messages.ts [hunk 2]
```diff
@@ -177,3 +177,8 @@ export const GOOGLE_RECAPTCHA_DOMAIN_ERROR =
 export const SERVER_API_TIMEOUT_ERROR =
   "Appsmith server is taking too long to respond. Please try again after some time";
 export const DEFAULT_ERROR_MESSAGE = "There was an unexpected error";
+
+export const LOCAL_STORAGE_QUOTA_EXCEEDED =
+  "Error saving a key in localStorage, you have exceeded the allowed storage size limit";
```
Comment (mohanarpit) [inline app/client/src/constants/messages.ts:182]:
```suggestion
  "Error saving a key in localStorage. You have exceeded the allowed storage size limit";
```

File: app/client/src/constants/messages.ts [hunk 3]
```diff
@@ -177,3 +177,8 @@ export const GOOGLE_RECAPTCHA_DOMAIN_ERROR =
...
+export const LOCAL_STORAGE_NO_SPACE_LEFT_ON_DEVICE =
+  "Error saving a key in localStorage, you have run out of disk space";
```
Comment (mohanarpit) [inline app/client/src/constants/messages.ts:184]:
```suggestion
  "Error saving a key in localStorage. You have run out of disk space";
```

File: app/client/src/constants/AppConstants.ts [hunk 4]
```diff
@@ -13,7 +15,7 @@ export const getAppStore = (appId: string) => {
   const appStoreName = getAppStoreName(appId);
   let storeString = "{}";
   // Check if localStorage exists
-  if (localStorage) {
+  if (window.localStorage) {
```
Comment (hetunandu) [inline app/client/src/constants/AppConstants.ts:18]: This check should happen in the utils function itself

**Review by hetunandu — APPROVED** [review]

**Comment by hetunandu** [comment]
Tests are failing

**Comment by rishabhsaxena** [comment]
> Tests are failing
@hetunandu tests are passing now

**Review by hetunandu — APPROVED** [review]

**Review by hetunandu — APPROVED** [review]

**Pull request merged**